libs/cardano-ledger-test/src/Test/Cardano/Ledger/STS.hs

```diff
@@ -68,6 +69,33 @@ stsPropertyV2 ::
   (env -> st -> sig -> st -> p) ->
   Property
 stsPropertyV2 specEnv specState specSig prop =
+  stsPropertyV2' @r specEnv specState specSig (\env _ _ -> specState env) prop
+
+stsPropertyV2' ::
+  forall r fn era env st sig fail p.
+  ( era ~ ConwayEra StandardCrypto
+  , Environment (EraRule r era) ~ env
+  , State (EraRule r era) ~ st
+  , Signal (EraRule r era) ~ sig
+  , PredicateFailure (EraRule r era) ~ fail
+  , STS (EraRule r era)
+  , BaseM (EraRule r era) ~ ReaderT Globals Identity
+  , PrettyA st
+  , PrettyA sig
+  , PrettyA env
+  , PrettyA fail
+  , Testable p
+  , HasSpec fn env
+  , HasSpec fn st
+  , HasSpec fn sig
+  ) =>
+  Specification fn env ->
+  (env -> Specification fn st) ->
+  (env -> st -> Specification fn sig) ->
+  (env -> st -> sig -> Specification fn st) ->
+  (env -> st -> sig -> st -> p) ->
```

Comment:
Might leave a comment about how the extra parameter of type 
(env -> st -> sig -> Specification fn st) ->

adds to this new kind of test,